aiosmtpd/smtp.py

```diff
@@ -322,6 +322,7 @@ def __init__(
             ident: Optional[str] = None,
             tls_context: Optional[ssl.SSLContext] = None,
             require_starttls: bool = False,
+            is_using_smtps: bool = False,
```

Comment:
After doing some tracing, a thought occurred to me:

Implicit TLS (SMTPS) is triggered if `tls_context` is set, right?

Can we just set `is_using_smtps` attribute from that? So `self._is_using_smtps = (tls_context is not None)`?

I might've overlooked something, so I welcome a discussion here.

Reply:
Well, yes and no.
For once, `tls_context` is for `STARTTLS`; `SMTPS` uses `ssl_context`. (This was already such before, I did not change it.)

Otherwise we _could_ set that, yep, but this option communicates _clear intent_.

As you've noted in #281, there is _no possible means for detecting_ whether or not we are using implicit TLS (`SMTPS`) within the `SMTP` class.

This goes to the extent that `ssl_context` is actually _not used by the `SMTP` class **at all**_.
It is still kept by me, as if later authentication methods will require e.g. client re-validation or something, it could happen.

But unlike `is_using_smtps`, `ssl_context` may or may not be an explicit intent.
`ssl_context` being `None` could equally mean "should not use SMTPS", as well as "hey, we/they/... failed to handle a key / cert error / expiry / ... somewhere".

But as per the above, there is _no means_ to detect whichever from inside the `SMTP` class; as such, if the described scenario happens, the client/user would just note that "hey, we don't have `AUTH`", or even "got `SMTP` error, pls help, it worked yesterday".

Currently, if `is_using_smtps` is set with no `ssl_context`, it should present an error.
I understand your point and will change it if that'd be the agreement, but I feel that would be a technically less correct approach.